src/Models/LoginModel.php

```diff
@@ -67,6 +67,20 @@ public function recordLoginAttempt(
         $this->checkQueryReturn($return);
     }
 
+    /**
+     * Returns the previous login information for the user,
+     * useful to display to the user the last time the account
+     * was accessed.
+     */
+    public function previousLogin(User $user): ?Login
+    {
+        $previous = $this->where('success', 1)
+            ->where('user_id', $user->id)
+            ->orderBy('id', 'desc')
+            ->findAll(1, 1);
+        return ($previous) ? $previous[0] : NULL;
```

Comment:
How about this?
```suggestion
        return $this->where('success', 1)
            ->where('user_id', $user->id)
            ->orderBy('id', 'desc')
            ->findAll(1, 1)[0] ?? null;
```
We could probably also use `limit()` with `first()`.

Reply:
Yes, I tried `findAll(1,1)[0]` but it returned an invalid offset type error for users with one or no login attempts.
I'll try with `limit()` and `first()` now.

Thank you.

Reply:
Done this.